tests/hardhat/Fork/pegStabilityTest.ts

```diff
@@ -0,0 +1,179 @@
+    let price: BigNumber;
+    describe("Peg Stability USDT", () => {
+      before(async () => {
+        defaultSigner = (await ethers.getSigners())[0];
+        psmUSDT = await deployPegStability(Contracts.vUSDT);
+        usdtSigner = await initMainnetUser(USDT_HOLDER);
+        vaiSigner = await initMainnetUser(VAI_HOLDER);
+        USDT = FaucetToken__factory.connect(Contracts.USDT, usdtSigner);
+        vaiAdmin = await initMainnetUser(Contracts.Timelock);
+        VAI = VAI__factory.connect(Contracts.VAI, vaiAdmin);
+        //Get oracle address
+        const comptroller = Comptroller__factory.connect(Contracts.Unitroller, defaultSigner);
+        const oracleAddress = await comptroller.oracle();
+        oracle = PriceOracle__factory.connect(oracleAddress, defaultSigner);
+        price = await oracle.getUnderlyingPrice(Contracts.vUSDT);
+        // Fund Timelock
+        await usdtSigner.sendTransaction({
+          to: await vaiAdmin.getAddress(),
+          value: convertToUnit(1, 18),
+        });
+        //Set PSM as VAI ward
+        await VAI.rely(psmUSDT.address);
+      });
+      describe("Initialization", () => {
+        it("Validate initialization parameters", async () => {
+          expect(await psmUSDT.vaiAddress()).to.equal(Contracts.VAI);
+          expect(await psmUSDT.comptroller()).to.equal(Contracts.Unitroller);
+          expect((await psmUSDT.vTokenAddress()).toLocaleLowerCase()).to.equal(Contracts.vUSDT.toLocaleLowerCase());
+          expect((await psmUSDT.stableTokenAddress()).toLocaleLowerCase()).to.equal(Contracts.USDT.toLocaleLowerCase());
+          expect((await psmUSDT.venusTreasury()).toLocaleLowerCase()).to.equal(venusTreasury);
+          expect(await psmUSDT.feeIn()).to.equal(feeIn);
+          expect(await psmUSDT.feeOut()).to.equal(feeOut);
+          expect(await psmUSDT.vaiMintCap()).to.equal(vaiMintCap);
+          expect(await psmUSDT.vaiMinted()).to.equal(0);
+          expect(await psmUSDT.isPaused()).to.be.false;
+          expect(await psmUSDT.accessControlManager()).to.equal(acmAddress);
+        });
+        it("Should not be able to re-initialize", async () => {
+          await expect(
+            psmUSDT.initialize(acmAddress, venusTreasury, Contracts.Unitroller, feeIn, feeOut, vaiMintCap),
+          ).to.be.rejectedWith("Initializable: contract is already initialized");
+        });
+      });
+      describe("Swap", () => {
+        it("USDT -> VAI", async () => {
+          const token_amount = BigNumber.from(convertToUnit(1000, 18));
+          const fee = token_amount.mul(feeIn).div(BASIS_POINT_DIVISOR);
+          const vai_to_mint = token_amount.sub(fee);
+          await USDT.connect(usdtSigner).approve(psmUSDT.address, token_amount);
+          const tx = await psmUSDT.connect(usdtSigner).swapStableForVAI(USDT_HOLDER, convertToUnit(1000, 18));
+          await expect(tx).to.emit(psmUSDT, "StableForVAISwapped").withArgs(token_amount, vai_to_mint, fee);
```

Comment:
Could we check the VAI balance of `usdtSigner` before and after the swap, and verify how much it was increased?